File: Difficulty: Medium/Implement Atoi/implement-atoi.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:       
    int myAtoi(char *s) {
        // Your code here
        int i=0;
        while(s[i]!='\0' && s[i]==' '){
            i++;
        }
        int sign = 1;
        if(s[i]=='+' || s[i]=='-'){
            sign=1-2*(s[i++]=='-');
        }
        int res = 0;
        while(s[i] >= '0' && s[i] <= '9' && s[i] != '\0'){
            if(res > INT_MAX/10 || (res == INT_MAX/10 && s[i]>'7')){
                return sign == 1 ? INT_MAX:INT_MIN;
            }
            res = res * 10 + (s[i++] -'0');
        }
        return res * sign;
    }
};


//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        char s[20];
        cin >> s;
        Solution ob;
        int ans = ob.myAtoi(s);
        cout << ans << endl;
        cout << "~" << endl;
    }
}
// } Driver Code Ends

File: Medium/Implement Atoi/implement-atoi.py
#User function template for Python

class Solution:
    # your task is to complete this function
    # function should return an integer
    def atoi(self,string):
        # Code here
        if string[0]=="-" and string[1:].isnumeric():
                return int(string)
        if string.isnumeric():
            return int(string)
        else:
            return -1





#{ 
 # Driver Code Starts
#Initial template for Python

if __name__=='__main__':
    t=int(input())
    for i in range(t):
        string = input().strip();
        ob=Solution()
        print(ob.atoi(string))
# } Driver Code Ends